lib/rules/curly.js

```diff
@@ -359,19 +421,23 @@ module.exports = {
                 check() {
                     if (this.expected !== null && this.expected !== this.actual) {
                         if (this.expected) {
+                            const loc = getReportingLoc(node, name);
+
                             context.report({
                                 node,
-                                loc: (name !== "else" ? node : getElseKeyword(node)).loc.start,
+                                loc,
```

Comment:
```suggestion
                                loc: body.loc,
```

I think there's no need for `function getReportingLoc`, we can just use the `body` param instead. That's the body/consequent/alternate node being currently checked.

Reply:
Yeah, now we can directly use `body.loc`, updated 👍🏻